src/components/karaoke.tsx

```diff
@@ -219,8 +220,20 @@ export function Karaoke({ code }: { code: string }) {
     return match ? match[1] : null;
   };
 
+  if (loading) {
+    return (
+      <div className="flex justify-center items-center h-screen">
+        Loading session...
+      </div>
+    );
+  }
+
+  if (error) {
+    return <div className="text-red-500 text-center">{error}</div>;
+  }
+
   return (
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

It's good to see error handling and loading states being managed. However, it would be better to provide more user-friendly messages instead of just "Loading session..." and displaying the raw error message. Consider adding a retry button for the error state.



<!-- This is an auto-generated comment by OSS CodeRabbit -->